trio/_core/tests/test_run.py

```diff
@@ -2195,3 +2196,33 @@ async def test_cancel_scope_deadline_duplicates():
             cscope.deadline = now + 9998
             cscope.deadline = now + 9999
         await sleep(0.01)
+
+
+@pytest.mark.skipif(
+    sys.implementation.name != "cpython", reason="Only makes sense with refcounting GC"
+)
+async def test_simple_cancel_scope_usage_doesnt_create_cyclic_garbage():
```

Comment:
👍 

Would it be worth adding a test to catch garbage regressions at a higher level, say `sleep(0)`?  Even if that case isn't at 0 now, we could assert on the count being below what's observed now.